tests/utils/splitAtom.test.tsx

```diff
@@ -375,3 +375,50 @@ it('no error on wrong atom configs (fix 510)', async () => {
   expect(numbersEl.textContent).toBe('0')
   expect(console.warn).toHaveBeenCalledTimes(1)
 })
+
+it('changing array of items (fix #736)', async () => {
+  const collectionAtom = atom<number[]>([])
+  const collectionAtomsAtom = splitAtom(collectionAtom)
+
+  const derivativeAtom = atom((get) => {
+    return get(collectionAtomsAtom).map(
+      (ca) => get(ca) + Math.round(Math.random() * 150)
```

Comment:
```suggestion
      (ca, index) => get(ca) + index
```
Can this be like this? Math.random() is a side effect, which is not generally allowed in the read function.